src/System.Security.SecureString/src/System/Security/SecureString.Unix.cs

```diff
@@ -0,0 +1,274 @@
+            }
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private void SetAtCore(int index, char c)
+        {
+            // Overwrite the character at the specified index
+            _buffer.Write((ulong)(index * sizeof(char)), c);
+        }
+
+        [System.Security.SecurityCritical]  // auto-generated
+        internal unsafe IntPtr ToUniStrCore()
+        {
+            int length = _decryptedLength;
+
+            byte* bufferPtr = null;
+            IntPtr stringPtr = IntPtr.Zero, result = IntPtr.Zero;
+            try
+            {
+                _buffer.AcquirePointer(ref bufferPtr);
+
+                // Allocate space for the string to be returned, including space for a null terminator
+                if ((stringPtr = Marshal.AllocCoTaskMem((length + 1) * sizeof(char))) == IntPtr.Zero)
+                    throw new OutOfMemoryException();
+
+                // Copy all of our data into it
+                Buffer.MemoryCopy(
+                    source:  bufferPtr, 
+                    destination:  (byte*)stringPtr.ToPointer(),
+                    destinationSizeInBytes: ((length + 1) * sizeof(char)),
+                    sourceBytesToCopy: length * sizeof(char));
+
+                // Add the null termination
+                *(length + (char*)stringPtr.ToPointer()) = '\0';
+
+                // Finally store the string pointer into our result.  We maintain
+                // a separate result variable to make clean up in the finally easier.
+                result = stringPtr;
+            }
+            finally
+            {
+                // If there was a failure, such that result isn't initialized, 
+                // release the string if we had one.
+                if (result == IntPtr.Zero && stringPtr != IntPtr.Zero)
+                {
+                    HGlobalBuffer.ZeroMemory((byte*)stringPtr, (ulong)(length * sizeof(char)));
+                    Marshal.FreeCoTaskMem(stringPtr);
+                }
+
+                if (bufferPtr != null)
+                    _buffer.ReleasePointer();
+            }
+
+            return result;
+        }
+
+        // -----------------------------
+        // ---- PAL layer ends here ----
+        // -----------------------------
+
+        private void EnsureCapacity(int capacity)
+        {
+            // Make sure the requested capacity doesn't exceed SecureString's defined limit
+            if (capacity > MaxLength)
+                throw new ArgumentOutOfRangeException("capacity", SR.ArgumentOutOfRange_Capacity);
+
+            // If we already have enough space allocated, we're done
+            if ((ulong)(capacity * sizeof(char)) <= _buffer.ByteLength)
+                return;
+
+            // We need more space, so allocate a new buffer, copy all our data into it,
+            // and then swap the new for the old.
+            HGlobalBuffer newBuffer = HGlobalBuffer.Allocate(capacity * sizeof(char));
```

Comment:
FWIW, this will probably end up being moot.  In order to use mlock to keep the pages from getting swapped out, I think I'll need to align on a page boundary and ensure that nothing else can be on the page, which will mean growing by increments of the page size.  I'll play around with it a bit later.